### PR TITLE
level support in reset_index

### PR DESCRIPTION
**Pull request by KathiaBarahona**

### CODE REVIEW

File: bach/tests/functional/bach/test_df_reset_index.py [hunk 1]
```diff
@@ -34,6 +34,24 @@ def test_reset_index_to_empty():
...
+    # level + drop
+    lbt = bt_cp.reset_index(level='city', drop=True)
+    assert 'city' not in lbt.data
+    assert list(lbt.index.keys()) == final_index
+
+    # dropping invalid level
+    invalid_level = 'random'
+    with pytest.raises(ValueError, match=fr"'{invalid_level}' level not found"):
+        bt_cp.reset_index(level=['city', invalid_level])
+
```
Comment (thijs-obj) [inline bach/tests/functional/bach/test_df_reset_index.py:54]: Can I suggest one more test case:
``` python
def test_reset_index_no_change():
    bt = get_bt_with_test_data()
    bt = bt.set_index(['skating_order', 'city'], append=True)
    lbt = bt.reset_index(level=[])
    assert list(lbt.index.keys()) == list(bt.index.keys())
```

This doesn't currently work, but I think we should make it pass. It works with Pandas

File: bach/bach/dataframe.py [hunk 2]
```diff
@@ -978,13 +978,19 @@ def rename(self, mapper: Union[Dict[str, str], Callable[[str], str]] = None,
         df._data = new_data
         return None if inplace else df
 
-    def reset_index(self, drop: bool = False, inplace: bool = False) -> Optional['DataFrame']:
+    def reset_index(
+        self,
+        level: Optional[Union[str, Sequence[str]]] = None,
```
Comment (thijs-obj) [inline bach/bach/dataframe.py:983]: I think this is good.

@hendrik-obj do you agree, or do we need to support integers here for the columns too?

**Review by thijs-obj — CHANGES_REQUESTED** [review]
One small change needed I think

File: bach/tests/functional/bach/test_df_reset_index.py (hunk 1 above)
Comment (KathiaBarahona) [reply bach/tests/functional/bach/test_df_reset_index.py:54]: thanks, nice catch!

**Review by thijs-obj — COMMENTED** [review]
Looks good to me. But let's wait for @hendrik-obj to reply too

**Review by hendrik-obj — APPROVED** [review]
lgtm!

**Review by thijs-obj — APPROVED** [review]

**Pull request merged**